realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1648,6 +1649,33 @@ public static boolean compactRealm(RealmConfiguration configuration) {
         return BaseRealm.compactRealm(configuration);
     }
 
+    /**
+     * If the Realm is a partially synchronized Realm, fetch and synchronize the objects of a given
+     * object type that match the given query (in string format).
+     *
+     * The results will be returned asynchronously in the callback.
+     *
+     * @param clazz the class to query.
+     * @param query string query.
+     * @param callback A callback used to vend the results of a partial sync fetch.
+     * @throws IllegalStateException if it is called from a non-Looper or {@link IntentService} thread.
+     */
+    @Beta
+    public <E extends RealmModel> void subscribeToObjects(Class<E> clazz, String query, PartialSyncCallback<E> callback) {
```

Comment:
Should this return a `RealmAsyncTask` instead? 
Or maybe we shouldn't worry about this for just a preview API?

Reply:
I guess that depends on if it should be cancellable

Reply:
We generally return a `RealmAsyncTask` when Java sends the query (via OkHttp) this will just call the ObjectStore to manage the creation of the `__ResultSets` and listening for changes in the query's status. `RealmAsyncTask` is not appropriate is this case